api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -212,7 +213,7 @@ private void saveToBackUp(String path, String jsonResponse) {
 		}
 		
 		String directoryStructure = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_DHIS2BACKUP_FOLDER
-		        + path.substring(0, path.lastIndexOf(File.separator));
+		        + path.substring(0, path.lastIndexOf(File.separator) == -1? path.length() - 1 :path.lastIndexOf(File.separator));
```

Comment:
@ahammi maybe this?
```java
import java.nio.file.Paths;

Paths.get(OpenmrsUtil.getApplicationDataDirectory(), DHISCONNECTOR_DHIS2BACKUP_FOLDER, path)
.toString()
```

Reply:
@kaweesi yes this a vetted Java lib that is working fine, please try it out :-)
Example:
```java
System.out.println( Paths.get("/path/to", "some", "/place", "somewhere/").toString() );
```